pyhik/hikvision.py

```diff
@@ -677,3 +677,149 @@ def update_attributes(self, event, channel, attr):
+
+    urls = [
+        '%s/ISAPI/Event/triggers' % root_url,
+        '%s/Event/triggers' % root_url,
+    ]
+
+    response = None
+    for url in urls:
+        try:
+            response = session.get(url, timeout=CONNECT_TIMEOUT)
+            if response.status_code == requests.codes.ok:
+                break
+        except (requests.exceptions.RequestException,
+                requests.exceptions.ConnectionError):
+            continue
+
+    if response is None or response.status_code != requests.codes.ok:
+        _LOGGING.warning('Unable to fetch event triggers from NVR')
+        session.close()
+        return events
+
+    try:
+        tree = ET.fromstring(response.text)
+    except ET.ParseError as err:
+        _LOGGING.error('Failed to parse event triggers XML: %s', err)
+        session.close()
+        return events
+
+    # Find namespace from root tag
+    namespace = ''
+    root_tag = tree.tag
+    if root_tag.startswith('{'):
+        namespace = root_tag.split('}')[0] + '}'
+
+    # Try different XML structures (camera vs NVR)
+    event_triggers = tree.findall('.//{0}EventTrigger'.format(namespace))
+
+    for trigger in event_triggers:
+        # Get event type
+        event_type_elem = trigger.find('{0}eventType'.format(namespace))
+        if event_type_elem is None or not event_type_elem.text:
+            continue
+
+        event_type = event_type_elem.text.lower()
+
+        # Skip videoloss as it's used for watchdog
+        if event_type == 'videoloss':
+            continue
+
+        # Get channel number
+        channel_num = 0
+        for channel_name in CHANNEL_NAMES:
+            channel_elem = trigger.find('{0}{1}'.format(namespace, channel_name))
+            if channel_elem is not None and channel_elem.text:
+                try:
+                    channel_num = int(channel_elem.text)
+                    break
+                except ValueError:
+                    continue
+
+        # Check if any valid notification method is configured
+        notification_list = trigger.find(
+            '{0}EventTriggerNotificationList'.format(namespace))
+        has_valid_notification = False
+
+        if notification_list is not None:
+            for notification in notification_list:
+                method_elem = notification.find(
+                    '{0}notificationMethod'.format(namespace))
+                if method_elem is not None and method_elem.text:
+                    if method_elem.text.lower() in {m.lower() for m in VALID_NOTIFICATION_METHODS}:
+                        has_valid_notification = True
+                        break
+
+        if has_valid_notification:
+            # Map to friendly name
+            friendly_name = SENSOR_MAP.get(event_type)
+            if friendly_name:
+                events.setdefault(friendly_name, []).append(channel_num)
+
+    session.close()
+    return events
+
+
+def inject_events_into_camera(camera, events):
```

Comment:
I don't have an issue with this, although i'm not sure it would really be needed if you expanded the existing event function as noted.